tests/system/common.py

```diff
@@ -126,6 +126,18 @@ def cluster_info(mom_name='marathon-user'):
         print("Marathon MoM not present")
 
 
+@retrying.retry(wait_fixed=1000, stop_max_attempt_number=5, retry_on_exception=ignore_exception)
+def wait_for_marathon_about():
+    """ When the cluster is starting, it might happen that there is some delay in:
+    - marathon leader registration with mesos
+    - admin router refreshing cache (every 30s)
+
+    To improve our chances everything happened before we hit the v2/info we retry here.
+    """
+    client = marathon.create_client()
+    return client.get_about()
```

Comment:
according to the docs, this function exists to ensure that the 1 marathon leader is registered and admin router cache is updated...   questions:
1. the about is a hit against `v2/info` which will return regardless of status or registration.   Should we have an endpoint that does what is documented?   or is there something in the /info JSON response we can wait on?

2. I'm not sure why admin router refresh is mentioned here... the admin router on a master is going to route to it's local marathon.   I don't believe it takes 30s for an end point to materialize.  I don't understand what this comment means.

Reply:
and how is it different than https://github.com/mesosphere/marathon/blob/4c222b13bc73718e250e8ddafdb1627f4fd9773f/tests/system/fixtures/__init__.py#L14-L20

Reply:
which is call for all functions:  https://github.com/mesosphere/marathon/blob/a5caab6ac148aa4a12e082123b15839d184c2a65/tests/system/test_marathon_root.py#L54

Reply:
That's a good point. `wait_for_service_endpoint` there is already waiting for marathon.

Reply:
1. we are calling marathon through service/marathon so through admin router
2. admin router returns HTTP 404 on service/marathon until it receives info from mesos that marathon registered - this is done every 30s, it is even visible in the AR logs

But yeah, I guess we CAN probably use wait_for_service_endpoint instead, good point.

Reply:
btw. it is called for all test functions, right? but not for the setup function...